src/Sylius/Bundle/CartBundle/EventListener/SessionCartSubscriber.php

```diff
@@ -75,8 +75,11 @@ public function onKernelResponse(FilterResponseEvent $event)
             return;
         }
 
-        if (null !== $cart && null !== $cart->getId()) {
-            $request->getSession()->set($this->sessionKeyName, $cart->getId());
+        if (null !== $cart && null !== $cart->getId() && null !== $cart->getChannel()) {
+            $request->getSession()->set(
+                sprintf('%s.%s', $this->sessionKeyName, $cart->getChannel()->getCode()),
```

Comment:
I think that cart is not aware of channel in `CartBundle`, this change should be reverted.